src/components/tooltip/NetworkMarker.js

```diff
@@ -76,20 +76,27 @@ function _markerDetailBytesNullable(label: string, value: ?number): * {
   );
 }
 
-const ALL_NETWORK_PROPERTIES_IN_ORDER = [
-  'startTime',
+const PRECONNECT_PROPERTIES_IN_ORDER = [
   'domainLookupStart',
   'domainLookupEnd',
   'connectStart',
   'tcpConnectEnd',
   'secureConnectionStart',
   'connectEnd',
+];
+
+const ALL_NETWORK_PROPERTIES_IN_ORDER = [
+  'startTime',
+  ...PRECONNECT_PROPERTIES_IN_ORDER,
   'requestStart',
   'responseStart',
   'responseEnd',
   'endTime',
 ];
 
+const REQUEST_PROPERTIES_IN_ORDER = ALL_NETWORK_PROPERTIES_IN_ORDER.slice();
```

Comment:
I do find this one confusing to reason with. I always have to look up the splice operation, could this be simplified?

Reply:
Ah, I couldn't find a better idea when I wrote it, `splice` is the right operation here. I still can't find a simpler alternative. _But_ I can add a comment that's explain this, I actually meant to add one and forgot.

Reply:
I wrote:
```js
/* For a marker with a preconnect phase, the second displayed diagram may only
 * contain these properties.
 * We use `splice` to generate this list out of the previous arrays, taking
 * ALL_NETWORK_PROPERTIES_IN_ORDER as source, then removing all the properties
 * of PRECONNECT_PROPERTIES_IN_ORDER.
 */
```